### PR TITLE
Disable Realm.delete* when using partial sync

### LINKED ISSUE #5780: `deleteAll` crashes under Partial Sync
```
io.realm.exceptions.RealmException: '__ResultSets' doesn't exist in current schema.
at io.realm.internal.ColumnIndices.getColumnInfo(ColumnIndices.java:112)
at io.realm.RealmSchema.getColumnInfo(RealmSchema.java:250)
at io.realm.ImmutableRealmSchema.get(ImmutableRealmSchema.java:43)
at io.realm.RealmSchema.getAll(RealmSchema.java:88)
at io.realm.BaseRealm.deleteAll(BaseRealm.java:615)
at io.realm.Realm.deleteAll(Realm.java:138)
at io.realm.objectserver.PartialSyncTests.lambda$clearTable$8$PartialSyncTests(PartialSyncTests.java:289)
at io.realm.objectserver.PartialSyncTests$$Lambda$8.execute(Unknown Source)
at io.realm.Realm.executeTransaction(Realm.java:1397)
at io.realm.objectserver.PartialSyncTests.clearTable(PartialSyncTests.java:288)
at java.lang.reflect.Method.invoke(Native Method)
at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
at android.support.test.internal.runner.junit4.statement.RunBefores.evaluate(RunBefores.java:80)
at android.support.test.internal.runner.junit4.statement.RunAfters.evaluate(RunAfters.java:61)
at io.realm.rule.RunInLooperThread$TestThread.run(RunInLooperThread.java:495)
at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:428)
at java.util.concurrent.FutureTask.run(FutureTask.java:237)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1133)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:607)
at java.lang.Thread.run(Thread.java:761)
```

### PR DESCRIPTION
**Pull request by cmelchior**
Fixes #5780 

Right now it isn't clear why anyone would use these methods in the first place when using partial sync. Also since the semantics of these methods are unclear, it is better to disable them with a proper message instead of a random crash.

This also means we punt the problem till someone complains and we can judge the use cases then.

### CODE REVIEW

**Comment by Zhuinden** [comment]
I must admit, this is the cleanest solution.

File: realm/realm-library/src/main/java/io/realm/BaseRealm.java [hunk 1]
```diff
@@ -64,8 +64,12 @@
             "This Realm instance has already been closed, making it unusable.";
     private static final String NOT_IN_TRANSACTION_MESSAGE =
             "Changing Realm data can only be done from inside a transaction.";
-    static final String LISTENER_NOT_ALLOWED_MESSAGE = "Listeners cannot be used on current thread.";
-
+    static final String LISTENER_NOT_ALLOWED_MESSAGE =
+            "Listeners cannot be used on current thread.";
+    static final String DELETE_NOT_SUPPORTED_UNDER_PARTIAL_SYNC =
+            "This API is not supported by partially " +
+            "synchronized Realms. Either unsubscribe using 'Realm.unsubscribeAsync()' or " +
```
Comment (nhachicha) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: Do we have an integration test that demonstrates this works? (unsubscribing will delete elements locally)

File: realm/realm-library/src/main/java/io/realm/BaseRealm.java [hunk 2]
```diff
@@ -606,11 +610,17 @@ public boolean isEmpty() {
 
     /**
      * Deletes all objects from this Realm.
+     * <p>
+     * If the Realm is a partially synchronized Realm, all subscriptions will be cleared as well.
```
Comment (nhachicha) [inline realm/realm-library/src/main/java/io/realm/BaseRealm.java]: Remove comment since this will throw If the Realm is partially synchronized

**Comment by nhachicha** [comment]
@cmelchior need to update the partial sync test to reflect the changes form this PR 
 see CI failures https://github.com/realm/realm-java/pull/5782#pullrequestreview-100396073
```
io.realm.objectserver.PartialSyncTests > clearTable[Nexus 5X - 6.0] FAILED 

	java.lang.IllegalStateException: This API is not supported by partially synchronized Realms. Either unsubscribe using 'Realm.unsubscribeAsync()' or delete the objects using a query and 'RealmResults.deleteAllFromRealm()'

	at io.realm.Realm.delete(Realm.java:1591)


```

**Pull request merged**